python/rikai/spark/functions/vision.py

```diff
@@ -133,3 +142,43 @@ def video_to_images(
         )
         for idx, img in enumerate(video_iterator)
     ]
+
+
+@udf(returnType=ImageType())
+def spectrogram_image(video, size: int = 224) -> Image:
+    import ffmpeg
+
+    """Applied ffmpeg filter to generate spectrogram.
```

Comment:
add a link explaining what a audio spectrogram is used for?

Reply:
This [KDNuggets post](https://www.kdnuggets.com/2020/02/audio-data-analysis-deep-learning-python-part-1.html) has some nice exposition on building models for audio classification. Practitioners often use the spectrogram to featurize an audio sample. Then pairing with CNNs, the audio classification task can be recast as an image classification one.

Reply:
@smellslikeml would it make sense to generate a content audio embedding using the spectrogram and see if that gives us a unique signal?

Reply:
A spectrogram image is a simple way to represent audio. People often use other audio/signal processing libraries, specialized filter banks, and even [learnable ones](https://ai.googleblog.com/2021/03/leaf-learnable-frontend-for-audio.html).  Fourier transforms have an advantage in allowing us to recover the original audio by [inversion](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.istft.html).